db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1734,7 +1728,8 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
       if (cfd->imm()->NumNotFlushed() != 0 || !cfd->mem()->IsEmpty() ||
           !cached_recoverable_state_empty_.load()) {
         flush_memtable_id = cfd->imm()->GetLatestMemTableID();
-        flush_req.emplace_back(cfd, flush_memtable_id);
+        FlushRequest req{{cfd, flush_memtable_id}};
```

Comment:
It's a bit hard for me to reason about restricting the flush to `flush_memtable_id` while at the same time allowing only one entry per CF in the flush queue. If there's a second manual flush to the same CF while the first is still queued, how do we know the second manual flush will make progress? It looks like `SchedulePendingFlush()` will skip scheduling the second one due to `cfd->queued_for_flush()`, but the one in the queue would not include the latest memtable. 

It looks like we are saved from the above worry because the memtable ID is ignored (?) in non-atomic flush, but still that feels a bit risky. I wonder if we should be more explicit here, passing the maximum possible value or something.

Reply:
In non-atomic flush, `FlushJob::max_memtable_id_` is nullptr, thus `PickMemtables()` will pick all memtables. In atomic flush, it is `&flush_memtable_id`, thus atomic flush will flush memtables up to this. We make this difference when we construct `FlushJob` objects. It is a little misleading and I will change it.